packages/framework/src/Component/UploadedFile/UploadedFileLocator.php

```diff
@@ -60,9 +72,12 @@ public function getAbsoluteUploadedFileFilepath(UploadedFile $uploadedFile)
     public function getUploadedFileUrl(DomainConfig $domainConfig, UploadedFile $uploadedFile)
     {
         if ($this->fileExists($uploadedFile)) {
-            return $domainConfig->getUrl()
-            . $this->uploadedFileUrlPrefix
-            . $this->getRelativeUploadedFileFilepath($uploadedFile);
+            $domainRouter = $this->domainRouterFactory->getRouter($domainConfig->getId());
+
+            return $domainRouter->generate('front_download_uploaded_file', [
+                'uploadedFileId' => $uploadedFile->getId(),
+                'uploadedFileName' => $uploadedFile->getSlug() . '.' . $uploadedFile->getExtension(),
```

Comment:
Do not we want to implement method `getSlugWithExtension()`? or the extension to be part of the name?

Reply:
Done